src/main/java/de/maxhenkel/peek/mixin/MultiPlayerGameModeMixin.java

```diff
@@ -0,0 +1,20 @@
+package de.maxhenkel.peek.mixin;
+
+import de.maxhenkel.peek.data.DataStore;
+import net.minecraft.client.multiplayer.MultiPlayerGameMode;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.inventory.ClickType;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+
+@Mixin(MultiPlayerGameMode.class)
+public class MultiPlayerGameModeMixin {
+    @Inject(method = "handleInventoryMouseClick", at = @At("TAIL"))
+    public void onHandleInventoryMouseClick(int i, int j, int k, ClickType clickType, Player player, CallbackInfo ci) {
+        if (DataStore.enderChestOpen) {
+            DataStore.ecinv = player.containerMenu.getItems();
```

Comment:
Contents should be also cached when opening and closing the container

Reply:
Contents are already stored on open using the onHandleContainerContent mixin. Can you clarify why we should also cache on close? An ender chest cannot be modified via any means except a player action. We record the state when the chest is opened for the first time in a session, and we record the state whenever a modification is made.

Reply:
It can be changed with other mods.